### DISCUSSION TITLE
Get Client Data and/or Header Data for each new connection

### DISCUSSION THREAD

**Discussion by cheak1974** [h2oai/wave#1751] [Q&A]
Hi Everybody,

I was wondering that I couldn't find a single word about client/session data like the IP Address of the connecting client in the docs.
I want to log the IP addresses of the visitors connecting to my site, but there doesn't seem to be a function for that.

How can I do it?

If there is a possibility maybe one could answer me a further question how custom header data could be read by h2o wave too. I plan to use h2o Wave behind a reverse proxy (haproxy). But if you connect through the reverse proxy the IP address of the connecting client would always be the IP address of the reverse proxy and not of the client. For that reason I configured haproxy to pack some custom header data into the packages:

```
http-request add-header X-Forwarded-For %[src]
http-request add-header X-Forwarded-Port %[dst_port]
http-request add-header Referer %[req.hdr(Referer)] 
```

Is it possible to get this header data inside the h2o wave apps?

regards
Chris

**Comment by mturoci** [comment]
> I was wondering that I couldn't find a single word about client/session data like the IP Address of the connecting client in the docs.

This is currently not possible as Wave server doesn't expose such low level data.

> how custom header data could be read by h2o wave

Not possible currently either, tracked in https://github.com/h2oai/wave/issues/1704.
